server/controllers/Messages/inputs.ts

```diff
@@ -0,0 +1,26 @@
+import { MaxLength } from 'class-validator';
+import { IsListEmpty } from './validators/isListEmpty';
+import { ValidateEmailList } from './validators/validateEmailList';
+import { FindDuplicateEmails } from './validators/findDuplicateEmails';
+import { InputType, Field } from 'type-graphql';
+
+// @TODO create custom validator to verify all emails belong to registered users
+// @TODO create custom validator to verify user sending email is authenticated
+
+@InputType()
+export class SendEmailInputs {
+  @Field(() => [String])
+  @IsListEmpty({ message: 'email list cannot be empty' })
+  @ValidateEmailList({ message: 'list contains invalid email' })
+  @FindDuplicateEmails({
+    message: 'list contains one or more duplicate emails',
+  })
+  to: [];
+
+  @Field()
+  @MaxLength(10)
```

Comment:
I set it to 10 characters so I didn't have to write out a long string in the tests and I wasn't sure what the appropriate limit should be, so I just left it at 10 for now. Any ideas on what the appropriate limit should be?

Reply:
If we're talking about an email address, here's a quote from the standard.

> In addition to restrictions on syntax, there is a length limit on email addresses. That limit is a maximum of 64 characters (octets) in the "local part" (before the "@") and a maximum of 255 characters (octets) in the domain part (after the "@") for a total length of 320 characters.

Reply:
I think it's the email subject and, according to https://stackoverflow.com/questions/1592291/what-is-the-email-subject-length-limit (and the RFC it references) the hard limit (per line) is 998 characters and the recommendation (again, per line) is 78.

I tested gmail and it has the limit of 998, so that might be a good idea.  Assuming that nodemailer can cope with that.

Reply:
By the way, there's a neat function to generate a long string.

```
var lotsOfAs = 'a'.repeat(999);
```